src/shared/helpers/check-user-role.ts

```diff
@@ -0,0 +1,17 @@
+import { Avo } from '@viaa/avo2-types';
+
+import { ShareRightsType } from '../components/ShareWithColleagues/ShareWithColleagues.types';
+import { Contributor } from '../types/contributor';
+
+export const checkUserRole = (
+	user: Avo.User.User,
+	subject: Avo.Assignment.Assignment | Avo.Collection.Collection,
+	contributors: Contributor[]
+): ShareRightsType => {
```

Comment:
for the name check... i would assume it returns a boolean that shows the result of the check
```suggestion
export const getUserContributorType = (
	user: Avo.User.User,
	subject: Avo.Assignment.Assignment | Avo.Collection.Collection,
	contributors: Contributor[]
): ShareRightsType => {
```